libkbfs/conflict_resolver.go

```diff
@@ -195,8 +206,16 @@ func (cr *ConflictResolver) Resolve(unmerged MetadataRevision,
 		return
 	}
 
+	cr.inputLock.Lock()
+	defer cr.inputLock.Unlock()
```

Comment:
In fact, I think even if the deadlock is solved, the bug remains. scenario would look like:

main goroutine: `Resolve` is called with rev=N, skips the cancel, gets to the send on `inputChan`.
bg goroutine: `processInput` receives on `inputChan`
main goroutine: `Resolve` is called with rev=N+1, skips the cancel, gets to the send on `inputChan` after releasing `inputLock`.
bg goroutine: `processInput` continues until spawning the worker goroutine, then receives N+1 from `inputChan`.
main goroutine: returns.
worker goroutine: does CR, clears the unmerged branch.
bg goroutine: `processInput` continues and cancels the previous worker, but it's too late.

Does that sound right?